### PR TITLE
Modify RealmsAdminResource.importRealm to work with InputStream

### LINKED ISSUE #13609: Modify `RealmsAdminResource.importRealm` to work with `InputStream`
Importing a realm via REST endpoint should not consume JSON realm representation but an `InputStream` instead.

The JSON realm representation is valid for the legacy store. The new store has to deal with either this representation or a native file with (likely) a file-store based structure.

This task is to prepare API so that the endpoint consumes `InputStream` and uses legacy / new store implementation logic to process the stream accordingly. This implies relevant changes in `RealmManager.importRealm` and `RepresentationToModel.importRealm`, pushing the conversion from the stream to representation as much to the storage layer as possible, and returning eventually the imported `RealmModel`.

### PR DESCRIPTION
**Pull request by ahus1**
Closes #13609

### CODE REVIEW

**Comment by ahus1** [comment]
Hi @hmlnarik - this is now ready for review. All builds are green. See below for notable changes:

* The request is now handed over as an InputStream to the repsective `ExportImportManager`s
* The endpoint is used by the UI not only to POST a full JSON, but also an almost empty JSON with just the realm name and if the realm should be enabled. This requires the `RealmManager.importRealm()` to replicate a lot of the logic from `createRealm()`. That needs to be preserved, as otherwise the UI won't be able to create a new realm. Therefore I decided not to replicate or break up the logic, instead there there's now an event fired by `MapExportImportManager` that also returns a value. In order have the event handled, I needed to create an SPI. I didn't see a way to do without one. If there is a simpler way, please let me know. 
* The error handling in the REST endpoints is fragile, as every handled exception need to set the transaction manager to rollback. This was missing for `ModelDuplicateException`, and I believe this is missing in a lot of other places where exceptions are caught in the endpoints. 
* I needed to change the `UncaughtErrorPageText` to use a different URL, as the error message for the endpoint changed from a specific 400 message to a more general 500 error as the handling is now done at a deeper level. I briefly considered adding custom import exceptions that could then be mapped to a specific REST response, but then discarded it as I consider the error handling in the REST endpoints too fragile and not worth the exception hierarchy. I might revisit it later. Another idea was to have exceptions that are mapped to REST responses automatically by using `WebApplicationException` as the parent, but discarded it as the datastore could shouldn't throw a REST specific exception. IMHO it would be nice to translate internal exceptions in the REST endpoint to exceptions that derive from `WebApplicationException`, as that would do the right thing with regards to transaction handling. But (again) this is out of scope for this issue.
* The new admin UI is currently broken and can't upload JSON when creating a new realm, see https://github.com/keycloak/keycloak-ui/issues/3316. Please use the old admin UI using the feature flags to test this manually.

As usual, please let me know if there are questions.

File: services/src/main/java/org/keycloak/services/managers/RealmManagerProvider.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+import org.keycloak.provider.Provider;
+
+/**
+ * This provider has been created to listen to {@link org.keycloak.storage.ImportRealmFromRepresentation} events.
+ * If that is no longer needed after further steps around the legacy storage migration, it can be removed.
+ *
+ * @author Alexander Schwartz
+ */
+public class RealmManagerProvider implements Provider {
```
Comment (hmlnarik) [inline services/src/main/java/org/keycloak/services/managers/RealmManagerProvider.java:28]: Please merge with `RealmManagerProviderFactory`.

Comment (ahus1) [reply services/src/main/java/org/keycloak/services/managers/RealmManagerProvider.java]: @hmlnarik - merged `RealmManagerProvider` into `RealmManagerProviderFactory`, so there is no longer a class named `RealmManagerProvider`. 

Local test successful, builds are still running. Will re-request a review once the build is complete.

**Comment by ahus1** [comment]
@hmlnarik - I added the `@Deprecated` annotation to the temporary event handling. 

All builds are green, it is now ready for the final review & merge IMHO.

**Review by hmlnarik — APPROVED** [review]
Approving on the condition that the `RealmManager` SPI is a temporary solution, as discussed in  a call

**Pull request merged**